pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -443,6 +443,7 @@ type GarbageCollector struct {
 	clock                            clock.Clock
 	registeredRateLimiter            *RegisteredRateLimiter
 	registeredRateLimiterForMonitors *RegisteredRateLimiter
+	deletionToWatchTracker           *uidToDeletionTime
```

Comment:
I'm not sure... The naming here is tricky, as it's clearly a deletionLatency from GC perspective, but it's not a deletionLatency from system perspective - I'm slightly worried that the plain name "deletionLatency" may be misleading (this of course applies only to the metric name - I don't care a lot about variable name, they probably should be consistent though).